src/dune_engine/fs_memo.ml

```diff
@@ -248,6 +249,7 @@ let path_kind path =
   | Error _ as error -> error
 
 let file_exists path =
+  let path = Path.as_outside_build_dir_exn path in
```

Comment:
Why are `file_exists` and a few other functions still taking `Path.t`? Why not switch the whole API to `Path.Outside_build_dir.t` consistently?

Reply:
It would be a much bigger change and perhaps the idea itself would be rejected. I can convert the rest of the api.